tests/unit/proxy/test_neo4j_proxy.py

```diff
@@ -622,6 +622,31 @@ def test_get_user_other_key_values(self) -> None:
             neo4j_user = neo4j_proxy.get_user(id='test_email')
             self.assertEqual(neo4j_user.other_key_values, {'mode_user_id': 'mode_foo_bar'})
 
+    def test_put_user_new_user(self) -> None:
+        """
+        Test creating a new user
+        :return:
+        """
+        with patch.object(GraphDatabase, 'driver') as mock_driver:
+            mock_session = MagicMock()
+            mock_driver.return_value.session.return_value = mock_session
+
+            mock_transaction = MagicMock()
+            mock_session.begin_transaction.return_value = mock_transaction
+
+            mock_run = MagicMock()
+            mock_transaction.run = mock_run
+            mock_commit = MagicMock()
+            mock_transaction.commit = mock_commit
```

Comment:
I'm not sure if this is a code consistency thing, but because the `mock_driver` is a mock object, you don't have to set the attributes to other mock objects (since it will create them dynamically). I believe you can do the following:
```suggestion
        with patch.object(GraphDatabase, 'driver') as mock_driver:
            mock_transaction = mock_driver.return_value.session.return_value.begin_transaction.return_value

            mock_run = mock_transaction.run
            mock_commit = mock_transaction.commit

            # Testing the mock below remains the same
```

Reply:
Indeed, good catch